### PR TITLE
⚡️ Sort portfolio items without calling additional metadata API

### PR DESCRIPTION
**Pull request by WeiJun0827**

### CODE REVIEW

File: src/store/modules/nft.js [hunk 1]
```diff
@@ -353,29 +362,72 @@ const actions = {
...
+
+    const nftClassIdDataMap = new Map();
+    collectedNFTs.forEach(n => nftClassIdDataMap.set(n.class_id, n.class_data));
+    createdNFTClasses.forEach(c => nftClassIdDataMap.set(c.id, c));
+    nftClassIdDataMap.forEach((classData, classId) => {
+      dispatch('formatAndSetNFTClassMetadata', { classId, classData });
     });
+
+    const formatedCreatedNFTClasses = createdNFTClasses.map(formatNFTClassInfo);
+    const formatedCollectedNFTs = collectedNFTs.map(formatNFTInfo);
```
Comment (nwingt) [inline src/store/modules/nft.js:379]:
```suggestion
    const formattedCreatedNFTClasses = createdNFTClasses.map(formatNFTClassInfo);
    const formattedCollectedNFTs = collectedNFTs.map(formatNFTInfo);
```

File: src/store/modules/nft.js [hunk 2]
```diff
@@ -254,28 +260,41 @@ const actions = {
...
       ...classMetadata,
       parent,
-      iscn_id: iscnId,
     };
+    commit(TYPES.NFT_SET_NFT_CLASS_METADATA, {
+      classId,
+      metadata: formattedMetadata,
+    });
+  },
+  async decorateNFTClassMetadataByURIAndISCN({ commit, getters }, classId) {
```
Comment (nwingt) [inline src/store/modules/nft.js:295]:
```suggestion
  async populateNFTClassMetadataFromURIAndISCN({ commit, getters }, classId) {
```

File: src/store/modules/nft.js [hunk 3]
```diff
@@ -254,28 +260,41 @@ const actions = {
...
     // const chainMetadata = await getClassInfo(classId);
     const { class: chainMetadata } = await this.$api.$get(
       api.getChainNFTClassMetadataEndpoint(classId)
     );
+    const { data, ...info } = chainMetadata;
+    const classData = { ...data, ...info };
+    dispatch('formatAndSetNFTClassMetadata', { classId, classData });
+    await dispatch('decorateNFTClassMetadataByURIAndISCN', classId);
+  },
+  formatAndSetNFTClassMetadata({ commit }, { classId, classData = {} }) {
```
Comment (nwingt) [inline src/store/modules/nft.js:275]:
```suggestion
  parseAndStoreNFTClassMetadata({ commit }, { classId, classData = {} }) {
```

**Review by nwingt — CHANGES_REQUESTED** [review]

File: src/store/modules/nft.js [hunk 4]
```diff
@@ -353,29 +362,72 @@ const actions = {
...
+
+    const formatedCreatedNFTClasses = createdNFTClasses.map(formatNFTClassInfo);
+    const formatedCollectedNFTs = collectedNFTs.map(formatNFTInfo);
     commit(TYPES.NFT_SET_USER_CLASSID_LIST_MAP, {
       address,
       nfts: {
-        created: normalizeNFTList(createdNFTClasses),
-        collected: normalizeNFTList(collectedNFTs),
+        created: normalizeNFTList(formatedCreatedNFTClasses, getters),
+        collected: normalizeNFTList(formatedCollectedNFTs, getters),
```
Comment (nwingt) [inline src/store/modules/nft.js:384]: I think normalizeNFTList can be part of the getters, so you dont have to include getters as params

**Review by nwingt — CHANGES_REQUESTED** [review]

File: src/store/modules/nft.js (hunk 4 above)
Comment (WeiJun0827) [reply src/store/modules/nft.js:384]: If `getNormalizeNFTList()` is a function from getters, can we use it before committing data into the store?

Comment (nwingt) [reply src/store/modules/nft.js:387]: I think it is no problem at all, and I think can keep `normalizeNFTList` for the name of the getter

File: src/store/modules/nft.js [hunk 5]
```diff
@@ -353,29 +363,61 @@ const actions = {
...
+    collectedNFTs.forEach(n => nftClassIdDataMap.set(n.class_id, n.class_data));
+    createdNFTClasses.forEach(c => nftClassIdDataMap.set(c.id, c));
+    nftClassIdDataMap.forEach((classData, classId) => {
+      dispatch('parseAndStoreNFTClassMetadata', { classId, classData });
     });
+
+    const formattedCreatedNFTClasses = createdNFTClasses.map(
+      formatNFTClassInfo
+    );
+    const formatedCollectedNFTs = collectedNFTs.map(formatNFTInfo);
```
Comment (nwingt) [inline src/store/modules/nft.js:382]:
```suggestion
    const formattedCollectedNFTs = collectedNFTs.map(formatNFTInfo);
```

**Review by nwingt — APPROVED** [review]

File: src/store/modules/nft.js [hunk 6]
```diff
@@ -239,6 +246,27 @@ const getters = {
     });
     return sorted;
   },
+  normalizeNFTList: (_, getters) => list =>
+    [
+      ...new Map(
+        [...list].map(({ classId, nftId, ...data }) => [
```
Comment (williamchong) [inline src/store/modules/nft.js:252]: why [...list] here? new Map and .map does not modify the array

File: src/store/modules/nft.js [hunk 7]
```diff
@@ -295,27 +337,16 @@ const actions = {
...
-        const iscnOwner = iscnRecord?.owner;
-        const iscnRecordTimestamp = iscnRecord?.records?.[0]?.recordTimestamp;
-        if (iscnOwner)
-          metadata = {
-            ...metadata,
-            iscn_owner: iscnOwner,
-            iscn_record_timestamp: iscnRecordTimestamp,
-          };
+        if (iscnRecord) {
+          metadata.iscn_owner = iscnRecord.owenr;
```
Comment (williamchong) [inline src/store/modules/nft.js:341]: owenr should be owner

File: src/util/misc.js [hunk 8]
```diff
@@ -9,4 +9,13 @@ export function getPropNameGenerator(propKey) {
   };
 }
 
+export function catchAxiosError(res) {
```
[the remark this answers is not in the document]
Comment (williamchong) [reply src/util/misc.js:12]: res mean response, but the param is actually unresolved promise

**Pull request merged**